### ISSUE TITLE
False positive RUF013 PEP 484 prohibits implicit `Optional`

### ISSUE THREAD

**Issue by randolf-scholz** [astral-sh/ruff#9334]
The bug occurs if `Optional` is imported from `typing_extensions` instead of `typing`.

```python
from typing_extensions import Optional

def foo(x: Optional[int] = None):
     pass
```

https://play.ruff.rs/dda2f645-171a-4666-93c6-d686c3f7edfa

**Comment by charliermarsh** [comment]
Fixed in https://github.com/astral-sh/ruff/pull/9335, thanks.
